pandas/core/internals/blocks.py

```diff
@@ -777,6 +777,13 @@ def _replace_coerce(
                 mask=mask,
             )
         else:
+            if value is None:
```

Comment:
why here instead of in 'replace'?

Reply:
because on main the recursion error occurs and we normally fix backports by opening PR against main and then backporting rather than against the backport branch directly.

Also, we would split the blocks in Block.replace which we didn't do on 1.3.5 and the regression fix restores previous behavior for now, see https://github.com/pandas-dev/pandas/issues/45601#issuecomment-1070935637.

I think is we do move to replace after the recursion is fixed we could also backport as a bug fix if we think that the block splitting is desirable to be consistent for 1.4.x

None handling is also slightly different in Block.replace than for a list-like so I suspect would need some other changes which happy as a followup on master.

This PR was a very targeted regression fix as a suitable backport for 1.4.x.

Reply:
that makes sense, thanks.